### PR TITLE
[BugFix] Fix week of the year behavior of date_format function inconsistent with mysql

### PR DESCRIPTION
**Pull request by zhangheihei**
Why I'm doing:
1 sr data_format behavior:
+----------------------------------+
| date_format('20241231', '%x-%v') |
+----------------------------------+
| 2025-01                          |
+----------------------------------+
the fe date_format behavior diff with be 
+-------------------------------+
| date_format('20241231', '%v') |
+-------------------------------+
| 53                            |
+-------------------------------+
2 mysql data_format behavior:
+-------------------------------+
| date_format('20241231', '%v') |
+-------------------------------+
| 01                            |
+-------------------------------+
What I'm doing:
fix the behavior
Fixes #issue

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [x] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by zhangheihei** [comment]
@trueeyu @imay 
Please help review code

**Comment by Seaven** [comment]
add a test case plz?

**Comment by zhangheihei** [comment]
> add a test case plz?

OK

**Comment by zhangheihei** [comment]
@Seaven  
hi Seaven, the UT has been added.

**Comment by zhangheihei** [comment]
@Seaven hi seaven, Please help review code

**Comment by zhangheihei** [comment]
@Seaven
hi seaven 
Quality Gate passed

**Review by Seaven — APPROVED** [review]

**Comment by Seaven** [comment]
> @Seaven hi seaven, Please help review code

Sorry, I'm late

File: fe/fe-core/src/test/java/com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctionsTest.java [hunk 1]
```diff
@@ -305,6 +305,14 @@ public void dateFormat() {
                 ScalarOperatorFunctions.dateFormat(ConstantOperator.createDate(LocalDateTime.of(2020, 2, 21, 13, 4, 5)),
                         ConstantOperator.createVarchar("asdfafdfsçv")).getVarchar());
 
+        Assert.assertNotEquals("53",
+                ScalarOperatorFunctions.dateFormat(ConstantOperator.createDatetime(LocalDateTime.of(2024, 12, 31, 22, 0, 0)),
+                        ConstantOperator.createVarchar("%v")).getVarchar());
+
+        assertEquals("01",
+                ScalarOperatorFunctions.dateFormat(ConstantOperator.createDatetime(LocalDateTime.of(2024, 12, 31, 22, 0, 0)),
+                        ConstantOperator.createVarchar("%v")).getVarchar());
```
Comment (silverbullet233) [inline fe/fe-core/src/test/java/com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctionsTest.java:314]: only keeping the second case is enough? seems these two are testing the same thing.

**Review by kangkaisen — APPROVED** [review]

File: fe/fe-core/src/test/java/com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctionsTest.java (hunk 1 above)
Comment (zhangheihei) [reply fe/fe-core/src/test/java/com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctionsTest.java:314]: hi, silver
These two cases are meant to explain the critical scenario

**Pull request merged**